include/aspect/boundary_traction/interface.h

```diff
@@ -126,31 +126,172 @@ namespace aspect
-     * @param factory_function A pointer to a function that can create an
-     * object of this traction boundary conditions model.
-     *
-     * @ingroup BoundaryTractions
-     */
-    template <int dim>
-    void
-    register_boundary_traction (const std::string &name,
-                                const std::string &description,
-                                void (*declare_parameters_function) (ParameterHandler &),
-                                std::unique_ptr<Interface<dim>> (*factory_function) ());
+        /**
+         * Declare the parameters of all known boundary traction plugins, as
+         * well as the ones this class has itself.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         * This determines which boundary traction objects will be created;
+         * then let these objects read their parameters as well.
+         */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * Go through the list of all boundary traction models that have been selected
+         * in the input file (and are consequently currently active) and return
+         * true if one of them has the desired type specified by the template
+         * argument.
+         */
+        template <typename BoundaryTractionType>
+        bool
+        has_matching_boundary_traction_model () const;
+
+        /**
+         * Go through the list of all boundary traction models that have been selected
+         * in the input file (and are consequently currently active) and see
+         * if one of them has the type specified by the template
+         * argument or can be casted to that type. If so, return a reference
+         * to it. If no boundary traction model is active that matches the given type,
+         * throw an exception.
+         */
+        template <typename BoundaryTractionType>
+        const BoundaryTractionType &
+        get_matching_boundary_traction_model () const;
+
+        /**
+         * For the current plugin subsystem, write a connection graph of all of the
+         * plugins we know about, in the format that the
+         * programs dot and neato understand. This allows for a visualization of
+         * how all of the plugins that ASPECT knows about are interconnected, and
+         * connect to other parts of the ASPECT code.
+         *
+         * @param output_stream The stream to write the output to.
+         */
+        static
+        void
+        write_plugin_graph (std::ostream &output_stream);
+
+
+        /**
+         * Exception.
+         */
+        DeclException1 (ExcBoundaryTractionNameNotFound,
+                        std::string,
+                        << "Could not find entry <"
+                        << arg1
+                        << "> among the names of registered boundary traction objects.");
+        /**
```

Comment:
Add an empty line here:
```suggestion

        /**
```